apps/hash-frontend/src/shared/use-actors.ts

```diff
@@ -0,0 +1,99 @@
+import { useQuery } from "@apollo/client";
+import {
+  currentTimeInstantTemporalAxes,
+  generateVersionedUrlMatchingFilter,
+  mapGqlSubgraphFieldsFragmentToSubgraph,
+  zeroedGraphResolveDepths,
+} from "@local/hash-isomorphic-utils/graph-queries";
+import {
+  blockProtocolPropertyTypes,
+  systemEntityTypes,
+} from "@local/hash-isomorphic-utils/ontology-type-ids";
+import { MachineProperties } from "@local/hash-isomorphic-utils/system-types/machine";
+import { AccountId, EntityRootType } from "@local/hash-subgraph";
+import { getRoots } from "@local/hash-subgraph/stdlib";
+import { useMemo } from "react";
+
+import { useUsers } from "../components/hooks/use-users";
+import {
+  StructuralQueryEntitiesQuery,
+  StructuralQueryEntitiesQueryVariables,
+} from "../graphql/api-types.gen";
+import { structuralQueryEntitiesQuery } from "../graphql/queries/knowledge/entity.queries";
+import { MinimalUser } from "../lib/user-and-org";
+
+export type MinimalActor =
+  | MinimalUser
+  | {
+      accountId: AccountId;
+      kind: "machine";
+      preferredName: string;
+    };
```

Comment:
Can we make this `displayName` to match the property type? Or are you trying to avoid typing headaches that might be caused by that?
```suggestion
  | {
      accountId: AccountId;
      kind: "machine";
      displayName: string;
    };
```

Reply:
Done in https://github.com/hashintel/hash/pull/3714/commits/53ddf2e361b24932e24b2e9709dff4622c71ef3b